webapp/components/PostTeaser/PostTeaser.vue

```diff
@@ -19,11 +20,37 @@
           <user-teaser :user="post.author" :group="post.group" :date-time="post.createdAt" />
           <hc-ribbon
             :class="[isPinned ? '--pinned' : '', post.image ? 'post-ribbon-w-img' : 'post-ribbon']"
-            :text="isPinned ? $t('post.pinned') : $t('post.name')"
+            :text="ribbonText"
+            :typ="post.postType[0]"
           />
         </div>
       </client-only>
       <h2 class="title hyphenate-text">{{ post.title }}</h2>
+      <ds-space
+        v-if="post && post.postType[0] === 'Event'"
```

Comment:
Here you work a lot with `if ... else` while this could be handled with a more general approach. (template...).

Furthermore I think you could avoid the filter backend-side for label post if you do not expect the postType to be an array of length 1....